sklearn/preprocessing/tests/test_data.py

```diff
@@ -851,6 +853,288 @@ def test_robust_scaler_iris_quantiles():
+                            n_quantiles=10,
+                            output_distribution='rnd').fit_transform, X)
+    assert_raises_regex(ValueError, "'output_distribution' has to be either"
+                        " 'norm' or 'uniform'. Got rnd instead.",
+                        QuantileTransformer(n_quantiles=10,
+                                            output_distribution='rnd').fit(
+                            X).inverse_transform, X)
+
+
+def test_quantile_transform_ignore_zeros():
+    X = np.array([[0, 0, 0, 0, 0],
+                  [1, 0, 2, 2, 1]]).T
+    X_sparse = sparse.csc_matrix(X)
+    transformer = QuantileTransformer(ignore_implicit_zeros=True,
+                                      n_quantiles=5)
+
+    # dense case -> warning raise
+    assert_warns_message(UserWarning, "'ignore_implicit_zeros' takes effect"
+                         " only with sparse matrix. This parameter has no"
+                         " effect.", transformer.fit, X)
+
+    X_gt = np.array([[0, 0, 0, 0, 0],
+                     [0, 0, 1, 1, 0]]).T
+    X_trans = transformer.fit_transform(X_sparse)
+    assert_almost_equal(X_gt, X_trans.A)
+
+    # consider the case where sparse entries are missing values and user-given
+    # zeros are to be considered
+    X_data = np.array([0, 0, 1, 0, 2, 2, 1, 0, 1, 2, 0])
+    X_col = np.array([0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+    X_row = np.array([0, 4, 0, 1, 2, 3, 4, 5, 6, 7, 8])
+    X_sparse = sparse.csc_matrix((X_data, (X_row, X_col)))
+    X_trans = transformer.fit_transform(X_sparse)
+    X_gt = np.array([[0., 0.5],
+                     [0., 0.],
+                     [0., 1.],
+                     [0., 1.],
+                     [0., 0.5],
+                     [0., 0.],
+                     [0., 0.5],
+                     [0., 1.],
+                     [0., 0.]])
+    assert_almost_equal(X_gt, X_trans.A)
+
+    transformer = QuantileTransformer(ignore_implicit_zeros=True,
+                                      n_quantiles=5)
+    X_data = np.array([-1, -1, 1, 0, 0, 0, 1, -1, 1])
+    X_col = np.array([0, 0, 1, 1, 1, 1, 1, 1, 1])
+    X_row = np.array([0, 4, 0, 1, 2, 3, 4, 5, 6])
+    X_sparse = sparse.csc_matrix((X_data, (X_row, X_col)))
+    X_trans = transformer.fit_transform(X_sparse)
+    X_gt = np.array([[0, 1],
+                     [0, 0.5],
+                     [0, 0.5],
+                     [0, 0.5],
+                     [0, 1],
+                     [0, 0],
+                     [0, 1]])
+    assert_almost_equal(X_gt, X_trans.A)
+    assert_almost_equal(X_sparse.A, transformer.inverse_transform(X_trans).A)
+
+
+def test_quantile_transform_dense_toy():
+    X = np.array([[0, 25, 50, 75, 100],
+                  [2, 4, 6, 8, 10],
+                  [2.6, 4.1, 2.3, 9.5, 0.1]]).T
+
+    transformer = QuantileTransformer(n_quantiles=5)
+    transformer.fit(X)
+
+    X_trans = transformer.fit_transform(X)
+    assert_almost_equal(np.min(X_trans, axis=0), 0.)
```

Comment:
Seeing as there are no duplicates in the input and capacity to represent the data in the transformer, can't you just check that the output corresponds to a linspace for each feature?